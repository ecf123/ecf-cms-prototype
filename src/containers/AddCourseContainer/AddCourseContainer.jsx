import "./AddCourseContainer.scss";
import { useState } from "react";
import AddCourse from "../../components/AddCourse/AddCourse";

const AddCourseContainer = () => {
  const [pageInfo, setPageInfo] = useState({
    file: null,
    fileName: "",
    fileTwo: null,
    fileTwoName: "",
    shortInputOne: "",
    shortInputTwo: "",
    freeTypeInputOne: "",
    freeTypeInputTwo: "",
  });

  const [errorMessage, setErrorMessage] = useState("");
  const [pageFirstOpened, setPageFirstOpened] = useState(true);

  const handleFileChange = (e) => {
    console.log(pageInfo)
      if (!pageInfo.file) setPageInfo({ fileName: e.target.files[0].name });
      pageInfo.file
        ? setPageInfo({ ...pageInfo, file: null })
        : setPageInfo({
            ...pageInfo,
            file: URL.createObjectURL(e.target.files[0]),
          });
    }

  const handleFileChange2 = (e) => {
    console.log(pageInfo)
    if (!pageInfo.fileTwo) setPageInfo({ fileTwoName: e.target.files[0].name });
      pageInfo.fileTwo
        ? setPageInfo({ ...pageInfo, fileTwo: null })
        : setPageInfo({
            ...pageInfo,
            fileTwo: URL.createObjectURL(e.target.files[0]),
          });
  }

  const handleChange = (e) => {
    console.log(e.target.name);
    setPageInfo({ ...pageInfo, [e.target.name]: e.target.value });
  };

  const storeInputs = (event) => {
    setPageFirstOpened(false);
    event.preventDefault();
    if (pageInfo.shortInputOne == "" || pageInfo.freeTypeInputOne == "" || pageInfo.shortInputTwo == "") {
      setErrorMessage("Please enter the info needed in the empty red boxes");
    } else {
      setErrorMessage("");
    }
  };

  
  const classNameInputTwo = (pageInfo.shortInputTwo != "" && !pageFirstOpened) ||
  (pageFirstOpened) 
  ? "short__input"
  : "short__input--empty"
  
  
  const classNameInput = (pageInfo.shortInputOne != "" && !pageFirstOpened) ||
  (pageFirstOpened) 
  ? "short__input"
  : "short__input--empty"
  
  const classNameInputThree = (pageInfo.freeTypeInputOne != ""  && !pageFirstOpened) ||
  (pageFirstOpened)
    ? "free-type__input"
    : "free-type__input--empty"

  const file= pageInfo.file

  const fileName= pageInfo.fileName

  const fileTwo = pageInfo.fileTwo

  const fileNameTwo = pageInfo.fileName

  const propsObject = {
    handleChange: handleChange,
    handleFileChange: handleFileChange,
    handleFileChange2: handleFileChange2,
    storeInputs: storeInputs,
    errorMessage: errorMessage,
    file: file,
    fileName: fileName,
    fileTwo: fileTwo,
    fileNameTwo: fileNameTwo,
    classNameInput: classNameInput,
    classNameInputTwo: classNameInputTwo,
    classNameInputThree: classNameInputThree,
  }
  
  return (
    <div className="add-course">
      <AddCourse
        shortLabelTextOne={"Course Name"}
        freeTypeLabelTextOne={"Course Overview"}
        shortLabelTextTwo={"Course Provider"}
        shortPlaceHolderTextTwo={"E.G Barclays..."}
        shortLabelTextThree={"Course Mentor"}
        shortPlaceHolderTextThree={"E.G Barclays"}
        uploadLabelName={"Course Thumbnail"}
        uploadLabelNameTwo={"Provider Logo"}
        propsObject={propsObject}
/>
    </div>
  );
};

export default AddCourseContainer;