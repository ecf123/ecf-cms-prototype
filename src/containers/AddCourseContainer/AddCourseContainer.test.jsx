import { customRender } from "../../utils/testUtils";
import AddCourseContainer from "./AddCourseContainer";

it("it should match the snapshot with props", () => {
    const { container } = customRender(<AddCourseContainer shortLabelTextOne={"Course Name"}/>);
    expect(container).toMatchSnapshot();
});

