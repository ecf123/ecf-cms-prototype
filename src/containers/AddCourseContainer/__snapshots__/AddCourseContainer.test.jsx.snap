// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match the snapshot 1`] = `
<div>
  <div
    class="add-course"
  >
    <div
      class="add-course"
    >
      <div
        class="add-course__input-texts"
      >
        <div
          class="short"
        >
          <label
            class="short__label"
          >
            Course Name
          </label>
          <input
            class="short__input"
            name="shortInputOne"
            spellcheck="true"
            type="text"
          />
        </div>
        <div
          class="free-type"
        >
          <label
            class="free-type__label"
          >
            Course Overview
          </label>
          <textarea
            class="free-type__input"
            name="freeTypeInputOne"
            spellcheck="true"
            type="text"
          />
        </div>
        <div
          class="short"
        >
          <label
            class="short__label"
          >
            Course Provider
          </label>
          <input
            class="short__input"
            name="shortInputTwo"
            placeholder="E.G Barclays..."
            spellcheck="true"
            type="text"
          />
        </div>
        <div
          class="short"
        >
          <label
            class="short__label"
          >
            Course Mentor
          </label>
          <input
            class="short__input"
            name="shortInputTwo"
            placeholder="E.G Barclays"
            spellcheck="true"
            type="text"
          />
        </div>
        <p
          class="add-course__error-message"
        />
        <div
          class="add-course__buttons"
        >
          <div
            class="add-course__cancel"
          >
            <button
              class="button light-grey round-border large"
            >
              <div />
              <p
                class="text large-text light-grey"
              >
                Cancel
              </p>
            </button>
          </div>
          <div
            class="add-course__submit"
          >
            <button
              class="button black round-border large"
            >
              <div />
              <p
                class="text large-text black"
              >
                Submit
              </p>
            </button>
          </div>
        </div>
      </div>
      <div
        class="add-course__uploadContainer"
      >
        <div
          class="add-course__media-upload"
        >
          <div
            class="thumbnail"
          >
            <label
              class="thumbnail__label"
              for="file1"
            >
              Course Thumbnail
            </label>
            <div
              class="thumbnail__container"
            >
              <div
                class="thumbnail__btn--wrapper"
              >
                <button
                  class="button light-grey round-border small"
                >
                  <div />
                  <p
                    class="text small-text light-grey"
                  >
                    Upload Image
                  </p>
                </button>
                <input
                  accept="image/*,video/*,audio/*,.txt"
                  class="thumbnail__btn--input"
                  id="file1"
                  type="file"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="add-course__media-upload2"
        >
          <div
            class="thumbnail"
          >
            <label
              class="thumbnail__label"
              for="file2"
            >
              Provider Logo
            </label>
            <div
              class="thumbnail__container"
            >
              <div
                class="thumbnail__btn--wrapper"
              >
                <button
                  class="button light-grey round-border small"
                >
                  <div />
                  <p
                    class="text small-text light-grey"
                  >
                    Upload Image
                  </p>
                </button>
                <input
                  accept="image/*,video/*,audio/*,.txt"
                  class="thumbnail__btn--input"
                  id="file2"
                  type="file"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
